src/SplitPackages/CsprojFileBuilder.cs

```diff
@@ -114,39 +113,79 @@ public void AddImports(string fx, string import)
 
         public void Execute()
         {
-            var document = new JObject();
-            document["dependencies"] = JObject.FromObject(CreateDependenciesDictionary(_dependencies));
-            document["frameworks"] = JObject.FromObject(CreateFrameworksDictionary());
-            document["runtimes"] = JObject.FromObject(CreateRuntimesDictionary());
-
-            var writer = _whatIf ? StreamWriter.Null : File.CreateText(_path);
-            using (var jsonWriter = new JsonTextWriter(writer) { Formatting = Formatting.Indented })
+            var document = new XDocument();
+            var dependenciesDictionary = CreateDependenciesDictionary(_dependencies);
+            var packageDependencies = CreatePackageReferenceList(dependenciesDictionary);
+            var root = new XElement("Project", new XAttribute("Sdk", "Microsoft.NET.Sdk"), new XAttribute("ToolsVersion", "15.0"));
+            var propertyGroup = new XElement("PropertyGroup");
+            propertyGroup.Add(new XElement("TargetFrameworks", string.Join(";", _frameworks.Select(i => Frameworks.GetMoniker(i.Name)))));
+
+            foreach (var framework in _frameworks)
             {
-                _logger.LogInformation($"Writing project.json file to {_path}");
-                document.WriteTo(jsonWriter);
+                var monikerName = Frameworks.GetMoniker(framework.Name);
+                if (!monikerName.StartsWith("netcoreapp"))
+                {
+                    var runtimes = Runtime.AllRuntimes.ToList();
+                    if (runtimes != null && runtimes.Count != 0)
+                    {
+                        propertyGroup.Add(new XElement("RuntimeIdentifiers", new XAttribute("Condition", $" '$(TargetFramework)' == '{monikerName}'"), string.Join(";", runtimes.Select(i => i.Name))));
+                    }
+                }
+
+                var frameworkDependencies = CreateFrameworksDictionary(framework.Dependencies);
+                var imports = framework.Imports;
+
+                if (imports != null && imports.Count != 0)
+                {
+                    propertyGroup.Add(new XElement("PackageTargetFallback", new XAttribute("Condition", $" '$(TargetFramework)' == '{monikerName}'"), $"$(PackageTargetFallback);{string.Join(";", imports.Select(i => Frameworks.GetMoniker(i)))}"));
+                }
+
+            }
 
-                if (_whatIf)
+            root.Add(propertyGroup);
+            root.Add(new XElement("ItemGroup", packageDependencies));
+            foreach (var framework in _frameworks)
+            {
+                var itemGroup = new XElement("ItemGroup");
+                var frameworkDependencies = CreateFrameworksDictionary(framework.Dependencies);
+                var monikerName = Frameworks.GetMoniker(framework.Name);
+                if (frameworkDependencies != null && frameworkDependencies.Count != 0)
```

Comment:
I would move the whole condition right before the item group and add it to the document only if there are dependencies specific to that framework.